### PR TITLE
quic: continuing implementation and cleanup

### PR DESCRIPTION
**Pull request by jasnell**
Continuing implementation...

##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: src/node_quic_socket.cc [hunk 1]
```diff
@@ -367,8 +361,54 @@ int QuicSocket::SendVersionNegotiation(
...
+    return -1;
+  }
+  QuicBuffer buf{NGTCP2_MAX_PKTLEN_IPV4};
+  ngtcp2_pkt_hd hd;
+
+  hd.version = chd->version;
+  hd.flags = NGTCP2_PKT_FLAG_LONG_FORM;
+  hd.type = NGTCP2_PKT_RETRY;
+  hd.pkt_num = 0;
+  hd.token = NULL;
```
Comment (danbev) [inline src/node_quic_socket.cc]: Can we make this `nullptr`?

File: src/node_quic_util.h [hunk 2]
```diff
@@ -36,25 +37,75 @@ constexpr uint16_t NGTCP2_APP_NOERROR = 0xff00;
...
+    switch (family) {
+       case AF_INET:
+        err = uv_ip4_addr(host, port, reinterpret_cast<sockaddr_in*>(addr));
+        break;
+      case AF_INET6:
+        err = uv_ip6_addr(host, port, reinterpret_cast<sockaddr_in6*>(addr));
+         break;
+       default:
+        CHECK(0 && "unexpected address family");
+        ABORT();
```
Comment (danbev) [inline src/node_quic_util.h]: We can remove this line as it is unreachable.

File: src/node_quic_util.h [hunk 3]
```diff
@@ -36,25 +37,75 @@ constexpr uint16_t NGTCP2_APP_NOERROR = 0xff00;
...
+constexpr size_t TOKEN_SECRETLEN = 16;
 constexpr size_t DEFAULT_MAX_STREAM_DATA_BIDI_LOCAL = 256_k;
 
 class SocketAddress {
  public:
+  static size_t GetMaxPktLen(const sockaddr* addr) {
+    return addr->sa_family ? NGTCP2_MAX_PKTLEN_IPV6 : NGTCP2_MAX_PKTLEN_IPV4;
+  }
+
+  static int ToSockAddr(
```
Comment (danbev) [inline src/node_quic_util.h]: This function looks quite similar to the [sockaddr_for_family](https://github.com/nodejs/node/blob/230f1f2aa4aa2d5800214c4894d1c13a3f153dd0/src/udp_wrap.cc#L177). Perhaps as some later stage we should have an util class that both can use.

Comment (jasnell) [reply src/node_quic_util.h]: let's consolidate later

**Comment by jasnell** [comment]
This is ready to go.

**Comment by jasnell** [comment]
Landed

**Pull request closed without merging**